fieldservice_vehicle_stock/models/stock_picking.py

```diff
@@ -11,26 +11,16 @@ class StockPicking(models.Model):
 
     @api.multi
     def action_assign(self):
-        res = {}
         for rec in self:
-            if rec.picking_type_id in (
-                # Vehicle Loading
-                self.env.ref(
-                    'fieldservice_vehicle_stock.'
-                    'picking_type_output_to_vehicle'),
-                # Location Pickup
-                self.env.ref(
-                    'fieldservice_vehicle_stock.'
-                    'picking_type_location_to_vehicle')
-            ):
+            if rec.picking_type_id.require_vehicle_id:
                 if rec.fsm_vehicle_id:
                     picking = \
                         rec.with_context(vehicle_id=rec.fsm_vehicle_id.id)
-                    res = super(StockPicking, picking).action_assign()
+                    res = super().action_assign()
                 else:
                     raise UserError(_(
                         "You must provide the vehicle for this picking type."))
-            res = super(StockPicking, rec).action_assign()
+            res = super().action_assign()
```

Comment:
We are calling call super() twice: on L19, and then again here.
Are you sure?

Reply:
@dreispt the second super() is for the if statement on line 15. If it does not go into that if statement then super() is not what we want.